src/tests/asyncTests.ts

```diff
@@ -162,12 +162,22 @@ export class AsyncTests {
       message: nls.localize('retrievingTestRunSummary')
     });
 
-    const testRunSummaryResults = (await this.connection.tooling.query(
-      testRunSummaryQuery,
-      {
-        autoFetch: true
-      }
-    )) as ApexTestRunResult;
+    let testRunSummaryResults;
+    try {
```

Comment:
@mingxuanzhangsfdx a couple of issues I see with the strategy of adding try catch on queries.

First, assuming all errors thrown by the query are due to out of memory exception will present the user will a false cause. Maybe the user's network fails some how causing a network exception or some other unexpected issue.

Second, out of memory errors cannot be at all.

Please research handling of out of heap space errors and then we can talk about what can be done.

Reply:
Agree with Pete. A not so very good approach is to look for specific message in strings but it is ver brittle and can break if the underlying error message returned changes